ThingIFSDK/ThingIFSDKLargeTests/ThingIFAPIScheduleTriggerTests.swift

```diff
@@ -0,0 +1,257 @@
+        self.onboardedApi.listTriggers(100)  { triggers, paginationKey, error in
+
+            { () in
+                XCTAssertNil(error)
+                XCTAssertNil(paginationKey)
+                guard let triggers = triggers else {
+                    XCTFail("triggers must not be nil")
+                    return
+                }
+
+                XCTAssertEqual(
+                  [triggerID1, triggerID2], triggers.map { $0.triggerID })
+
+
+                XCTAssertEqual(
+                  [checkDataForTemperature, checkDataForHumidity],
+                  (triggers.map { TriggerToCheck($0)! })
+                )
+
+            }()
+            expectation.fulfill()
+        }
+        self.waitForExpectations(timeout: TEST_TIMEOUT) { error in
+            XCTAssertNil(error)
+        }
+
+        expectation = self.expectation(description: "delete trigger")
+        self.onboardedApi.deleteTrigger(triggerID1) { deleted, error in
+            XCTAssertNil(error)
+            XCTAssertEqual(triggerID1, deleted)
+            expectation.fulfill()
+        }
+        self.waitForExpectations(timeout: TEST_TIMEOUT) { error in
+            XCTAssertNil(error)
+        }
+
+        let modifiedHumidityAliasActions = [
+          AliasAction(ALIAS2, actions: [Action("setPresetHumidity", value: 55)])
+        ]
+
+        let schedulePredicateForModifiedHumidity =
+          SchedulePredicate("* 1 * * *")
+        let checkDataForModifiedHumidity = TriggerToCheck(
+          true,
+          targetID: targetID,
+          enabled: true,
+          command: CommandToCheck(
+            false,
+            targetID: targetID,
+            issuerID: self.onboardedApi.owner.typedID,
+            commandState: nil,
+            hasFiredByTriggerID: false,
+            hasCreated: false,
+            hasModified: false,
+            aliasActions: modifiedHumidityAliasActions,
+            aliasActionResults: []
+          ),
+          predicate: schedulePredicateForModifiedHumidity)
+
+        expectation = self.expectation(description: "patch trigger")
+        self.onboardedApi.patchTrigger(
+            triggerID2,
+            triggeredCommandForm: TriggeredCommandForm(
+              modifiedHumidityAliasActions),
+            predicate: schedulePredicateForModifiedHumidity) { trigger, error in
+
+            XCTAssertNil(error)
+            XCTAssertEqual(
+              checkDataForModifiedHumidity,
+              TriggerToCheck(trigger))
+            expectation.fulfill()
+        }
+        self.waitForExpectations(timeout: TEST_TIMEOUT) { error in
+            XCTAssertNil(error)
+        }
+
+        expectation = self.expectation(description: "list trigger second")
+        self.onboardedApi.listTriggers(100) { triggers, paginationKey, error in
+
+            { () in
+                XCTAssertNil(error)
+                XCTAssertNil(paginationKey)
+                guard let triggers = triggers else {
```

Comment:
@rizakii 

I want to use `triggers` as non optional so I used `guard`. `XCTAssertNotNil` does not unwrap optional value.